core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceInfo.java

```diff
@@ -60,6 +68,15 @@ void setAdTrackingEnabled(boolean enabled) {
 
     void setAdvertisingId(String advertisingId) {
         this.advertisingId = advertisingId;
+        this.adTrackingEnabled = (this.advertisingId != null);
```

Comment:
Let's simplify this:
```suggestion
        this.adTrackingEnabled = true;
```

Reply:
updated